src/main/webapp/script.js

```diff
@@ -220,51 +220,117 @@ function getGraphDisplay(graphNodes, graphEdges) {
     node.tip.show();
   });
 
-  const searchElement = document.getElementById('search');
-  document.getElementById('search-button').onclick = function() {
-    if (searchNode(cy, searchElement.value) || searchElement.value == "") {
-      document.getElementById('search-error').innerText = "";
+  document.getElementById('reset').onclick = function(){ resetElements(cy) };
+
+  document.getElementById('search-button').onclick = function() { search(cy, "node", searchNode) };
+
+  document.getElementById('search-token-button').onclick = function() { search(cy, "token", searchToken) };
+}
+
+/**
+ * Searches based on type (node or token)
+ */
+function search(cy, type, searchFunction) {
+  resetElements(cy);
+  let errorText = "";
+  let query = document.getElementById(type + '-search').value;
+  let result;
+  if (query != "") {
+    result = searchFunction(cy, query);
+    if (result) {
+      errorText = "";
```

Comment:
isn't this what errorText is right now? have you considered doing
`if (!result) { // what you have down there}`